### DISCUSSION TITLE
Enable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [ropali/dockyard#18] [Ideas]
Hi!

I noticed that in the `Cargo.toml` [file](https://github.com/ropali/dockyard/blob/main/src-tauri/Cargo.toml) for the Rust backend Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit (not the case for the application but anyway - we are blazingly fast here).

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

In the most basic way, it can be enabled with the following lines added to the Cargo.toml:

```
[profile.release]
lto = true
```

Thank you.

**Comment by ropali** [comment]
Thanks for this amazing write-up. I will definitely look into it. thanks again.

**Comment by ropali** [comment]
Added it
